### PR TITLE
gh-94440: Fix issue of ProcessPoolExecutor shutdown hanging

### PR DESCRIPTION
**Pull request by yonatanp**
Added a test to reproduce the issue and a fix to the issue itself.

* Issue: gh-94440

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2022-06-30-21-28-41.gh-issue-94440.LtgX0d.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fixed ``concurrent.futures.process._ExecutorManagerThread`` to flush canceled
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2022-06-30-21-28-41.gh-issue-94440.LtgX0d.rst]: It's best not to discuss private, internal classes like `_ExecutorManagerThread` in news entries -- users of `concurrent.futures` generally won't care about such things. Try to frame the news entry in terms of the user-visible impact of the change instead.

**Comment by AlexWaygood** [comment]
@gpshead, would you be able to take a look at this PR?

File: Misc/NEWS.d/next/Library/2022-06-30-21-28-41.gh-issue-94440.LtgX0d.rst (hunk 1 above)
Comment (yonatanp) [reply Misc/NEWS.d/next/Library/2022-06-30-21-28-41.gh-issue-94440.LtgX0d.rst]: Makes sense - I rewrote the news item in that spirit.

Comment (AlexWaygood) [reply Misc/NEWS.d/next/Library/2022-06-30-21-28-41.gh-issue-94440.LtgX0d.rst]: Thanks! It looks like the latest commit has been authored by a GitHub account associated with a different email address. That's caused the CLA check to start to fail -- you may need to sign it with both email addresses in order to appease the bot (and our legal team) :)

**Comment by AlexWaygood** [comment]
(We prefer to avoid force pushes in the CPython repo, where possible -- they interact badly with the GitHub UI, making it hard to see what changed between commits. All PRs are squash-merged in CPython anyway, so there's no need to worry about having a messy commit history in this PR :)

**Comment by yonatanp** [comment]
> (We prefer to avoid force pushes in the CPython repo, where possible -- they interact badly with the GitHub UI, making it hard to see what changed between commits. All PRs are squash-merged in CPython anyway, so there's no need to worry about having a messy commit history in this PR :)

Yeah - totally agree. I had the unfortunate case where my news change commit was created using my work-account, instead of my personal account. This in turn blocked the merge as the CLA Signing check would fail. I'm counting on the squash to clear it all out in the end :)

**Comment by yonatanp** [comment]
:sigh: I'm creating a bit of a mess with the news item thing.
I might need some expert help here to get it back on track.
The check of generated files up-to-date was failing, and I assumed it was because I edited the news item manually (where initially it was created by the blurb-it bot). So I created a new blurb with the bot, but now I have two news items. I will manually delete the new one and commit.
Still not sure what's then the issue with the failing "generated files" check.

**Comment by yonatanp** [comment]
> Probably just a fluke. If it happens again, ping me and I'll take a look.

Failed again. It seems to be consistent.

**Comment by AlexWaygood** [comment]
> > Probably just a fluke. If it happens again, ping me and I'll take a look.
> 
> Failed again. It seems to be consistent.

It appears as though the CI check is broken somehow. It is failing on all PRs currently, e.g.:

- #102734

**Comment by AlexWaygood** [comment]
All green again now @yonatanp!

**Review by gpshead — APPROVED** [review]
Nice work debugging this. This looks good to me.

**Pull request merged**